network.go

```diff
@@ -68,9 +68,26 @@ const (
 	defaultQueues     = 8
 )
 
-type netIfaceAddrs struct {
-	iface net.Interface
-	addrs []net.Addr
+// NetIfaceAddrs describes a network interface an its associated IP addresses.
+type NetIfaceAddrs struct {
+	Iface net.Interface
+	Addrs []net.IPNet
+}
+
+// DNSInfo describes the DNS setup related to a network interface.
+type DNSInfo struct {
+	Servers  []string
+	Domain   string
+	Searches []string
+	Options  []string
+}
+
+// NetworkInfo gathers all information related to a network interface.
+// It can be used to store the description of the underlying network.
+type NetworkInfo struct {
+	Iface  NetIfaceAddrs
```

Comment:
Same here. Maybe we can switch the netlink.Link
This has another benefit. This will allow us to make interconnection decisions based on the type of the link, which we cannot do today without an extra scan.

For example if the interface is of type tap, macvtap we can directly connect it.